### PR TITLE
Cloud Audit Log Catalog

### PR DESCRIPTION
**Pull request by grant**
## Cloud Audit Log Catalog

Creates a catalog of Cloud Audit Log `serviceName` and `methodName` combinations using the audit catalog.
Required for `google.cloud.audit.log.v1.written` type.

### Preview

- https://github.com/googleapis/google-cloudevents/blob/grant_cal_catalog/AUDIT_CATALOG.md
- https://github.com/googleapis/google-cloudevents/blob/grant_cal_catalog/README.md#google-cloudevent-catalogs

### PR Changes

- Creates new file `AUDIT_CATALOG.md`.
  - This file is generated from data in `json/audit/service_catalog.json`. Note: that file is sync'd from g3 via copybara.
  - Updates top-level `README.md` to link to 2 event catalogs.
- Updates `tools/readme-catalog` tool
  - Splits catalog tool to include both the CloudEvent catalog and CAL catalog generators (and re-use common logic)
    - Creates index file for running generators
  - Create new `common` util file for catalog generators.
  - Changes tool to use ESM modules for easier imports.
- Includes detailed comments.

### CODE REVIEW

File: tools/readme-catalog/gen-audit-catalog.js
Comment (jskeet) [inline tools/readme-catalog/gen-audit-catalog.js:10]: Any reason this isn't actual valid JSON (with double quotes around property names, and double quotes instead of single quotes for values)?

File: tools/readme-catalog/gen-audit-catalog.js
Comment (jskeet) [inline tools/readme-catalog/gen-audit-catalog.js:1]: Copyright statement? (I realise we're somewhat inconsistent for scripts...)

**Review by jskeet — APPROVED** [review]
I'm not really competent to review JavaScript, but it all looks pretty sensible to me.

File: tools/readme-catalog/gen-audit-catalog.js
Comment (grant) [reply tools/readme-catalog/gen-audit-catalog.js:10]: Ah, this was taken from JS. Yeah, JSON would be easier. Will change.

File: tools/readme-catalog/gen-audit-catalog.js
Comment (grant) [reply tools/readme-catalog/gen-audit-catalog.js:1]: Yeah, will add. Good point.

Maybe there's a bot out there.

Comment (grant) [reply tools/readme-catalog/gen-audit-catalog.js:1]: Will be enforced with https://github.com/googleapis/google-cloudevents/pull/159

**Comment by grant** [comment]
Blocked by broken build.

Fixed with: https://github.com/googleapis/google-cloudevents/pull/162

File: tools/readme-catalog/gen-audit-catalog.js
Comment (grayside) [inline tools/readme-catalog/gen-audit-catalog.js:21]: Discussion: This is fantastically clear and detailed commenting. Is this complying with any particular commenting standard? I've been meaning to look up best practice around node.js.

File: tools/readme-catalog/gen-audit-catalog.js
Comment (grayside) [inline tools/readme-catalog/gen-audit-catalog.js:8]: In looking over the code in this file, it's not clear to me how this input is used.

File: tools/readme-catalog/gen-audit-catalog.js
Comment (grayside) [inline tools/readme-catalog/gen-audit-catalog.js:45]: nit: incomplete sentence

File: tools/readme-catalog/gen-cloudevent-catalog.js
Comment (grayside) [inline tools/readme-catalog/gen-cloudevent-catalog.js:23]: nit: A direct replace statement might be more clear.

```suggestion
  ).replace('.', '/');
```

**Review by grayside — CHANGES_REQUESTED** [review]
I left a couple nits and a couple action items.

Overall this makes sense.

Future Consideration: It seems like it's starting to push the limits of easily maintained "string hacking", keeping all these pieces in mind as an intermittent maintainer starts to get more challenging.

We should consider at what point using another approach such as a templating engine might be more maintainable.

File: README.md
Comment (grant) [reply README.md:42]: Applied.

File: README.md
Comment (grant) [reply README.md:47]: Thanks for the comment and suggestion. I've applied many of the suggested changes in the latest commit.

File: tools/readme-catalog/gen-audit-catalog.js
Comment (grant) [reply tools/readme-catalog/gen-audit-catalog.js:21]: Thank you.

I am generally using the JSDoc standard, which has support with editors like VS Code:

https://jsdoc.app/
https://code.visualstudio.com/docs/languages/javascript#_jsdoc-support

File: tools/readme-catalog/gen-audit-catalog.js
Comment (grant) [reply tools/readme-catalog/gen-audit-catalog.js:8]: I've updated the comment to make it clear how this input is used.

File: tools/readme-catalog/gen-audit-catalog.js
Comment (grant) [reply tools/readme-catalog/gen-audit-catalog.js:45]: Fixed.

File: tools/readme-catalog/gen-cloudevent-catalog.js
Comment (grant) [reply tools/readme-catalog/gen-cloudevent-catalog.js:23]: This replacement would result in an error.

`String.prototype.replace` only catches the first match.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace

In this case, we want to replace all `.` with `/` characters in the schema path. I've usually seen this pattern (change the substring to a simple regex) works.

**Review by grant — COMMENTED** [review]
Thanks for the review @grayside. I've replied to the comments. PTAL :)

**Review by grayside — APPROVED** [review]

**Pull request merged**
